File: go/orderly-queue.go
func orderlyQueue(s string, k int) string {
    if k == 1 {
        res := s
        double := s + s
        for i, _ := range s { 
            tmp := double[i : i + len(s)]
            if tmp < res {
                res = tmp
            }
        }
        return res
    }    
    chars := strings.Split(s, "")
    sort.Strings(chars)
    return strings.Join(chars, "")
}

File: rust/orderly-queue.rs
use std::cmp;
impl Solution {
    pub fn orderly_queue(s: String, k: i32) -> String {
        if k == 1 {
            let mut res = s.clone();
            let double: String = [s.clone(), s.clone()].join(""); 
            for i in 0..s.len() {
                let tmp = &double[i..i + s.len()];
                let tmp : String = tmp.into();
                res = cmp::min(res, tmp).clone();
            }
            
            res
        } else {
            let mut characters: Vec<char> = s.chars().collect();
            characters.sort();
            
            characters.iter().collect()
        }   
    }
}

// sugyan's solution
// https://leetcode.com/problems/orderly-queue/discuss/1446287/Rust-solution
impl Solution {
    pub fn orderly_queue(s: String, k: i32) -> String {
        if k == 1 {
            let ss = s.chars().chain(s.chars()).collect::<Vec<_>>();
            let mut v = ss.windows(s.len()).collect::<Vec<_>>();
            v.sort();
            v[0].iter().copied().collect()
        } else {
            let mut v = s.chars().collect::<Vec<_>>();
            v.sort();
            v.iter().collect()
        }
    }
}

